src/performance.js

```diff
@@ -0,0 +1,184 @@
+/**
+ * @license
+ * This file is part of the Game Closure SDK.
+ *
+ * The Game Closure SDK is free software: you can redistribute it and/or modify
+ * it under the terms of the Mozilla Public License v. 2.0 as published by Mozilla.
+
+ * The Game Closure SDK is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Mozilla Public License v. 2.0 for more details.
+
+ * You should have received a copy of the Mozilla Public License v. 2.0
+ * along with the Game Closure SDK.  If not, see <http://mozilla.org/MPL/2.0/>.
+ */
+
+ /**
+ * @module performance
+ *
+ * Calculates device performance score by maintaining a window history of the
+ * worst tick deltas and taking its average. 
+ *
+ * Over time if the window of worst ticks is not updated, then it will periodically
+ * pop off the worst deltas within the window for the performance score to rise back up.
+ *
+**/
+
+var DEFAULT_RANK = 0;
+var DEFAULT_ALLOW_REDUCTION = true;
+var DEFAULT_FPS_LOWER_BOUND = 10;
+var DEFAULT_FPS_UPPER_BOUND = 60;
+var MAX_WORST_TICKS_AMOUNT = 30;
+var MAX_SCORE = 100;
+var MIN_SCORE = 0;
+var TICKS_TIL_INCREASE_SCORE = 200;
+  
+var Performance = Class(function () {
+  var _ticksSinceLastWorstUpdate = 0;
+  var _worstTicks = [];
+  var _minFPS = DEFAULT_FPS_LOWER_BOUND;
+  var _maxFPS = DEFAULT_FPS_UPPER_BOUND;
+  var _canMeasure;
+  var _lastTick;
+
+  function _mapFPSToPerformanceScore (fps) {
+    var range1 = _maxFPS - _minFPS;
```

Comment:
These variables should be more clear:
`range1` -> `fpsRange`
`range2` -> `scoreRange`